New Monitor Metrics Query Library (#46194)
Azure/azure-sdk-for-java: sdk/monitor/azure-monitor-query-metrics/TROUBLESHOOTING.md

> # Troubleshooting Azure Monitor Query client library issues

Reviewer: ```suggestion
# Troubleshooting Azure Monitor Metrics Query client library issues
```

This troubleshooting guide contains instructions to diagnose frequently encountered issues while using the Azure
Monitor Query client library for Java.

## Table of contents

* [General Troubleshooting](#general-troubleshooting)
    * [Enable client logging](#enable-client-logging)
    * [Enable HTTP request/response logging](#enable-http-requestresponse-logging)

>     * [Troubleshooting authentication issues with logs and metrics query requests](#authentication-errors)

Reviewer: Do not refer to `logs query` in this README.

* [Troubleshooting Metrics Query](#troubleshooting-metrics-query)
    * [Troubleshooting authorization errors](#troubleshooting-authorization-errors-for-metrics-query)
    * [Troubleshooting unsupported granularity for metrics query](#troubleshooting-unsupported-granularity-for-metrics-query)

## General Troubleshooting

### Enable client logging

To troubleshoot issues with Azure Monitor query library, it is important to first enable logging to monitor the
behavior of the application. The errors and warnings in the logs generally provide
useful insights into what went wrong and sometimes include corrective actions to fix issues.
The Azure client libraries for Java have two logging options:

* A built-in logging framework.
* Support for logging using the [SLF4J](https://www.slf4j.org/) interface.

Refer to the instructions in this reference document on how to [configure logging in Azure SDK for Java](https://docs.microsoft.com/azure/developer/java/sdk/logging-overview).

### Enable HTTP request/response logging

Reviewing the HTTP request sent or response received over the wire to/from the Azure Monitor service can be useful in

> troubleshooting issues. To enable logging the HTTP request and response payload, the LogsQueryClient and the

Reviewer: Remove references to LogsQueryClient.

HTTP request/response.

Environment variable name: `AZURE_HTTP_LOG_DETAIL_LEVEL`

| Value            | Logging level                                                        |
|------------------|----------------------------------------------------------------------|
| none             | HTTP request/response logging is disabled                            |
| basic            | Logs only URLs, HTTP methods, and time to finish the request.        |
| headers          | Logs everything in BASIC, plus all the request and response headers. |
| body             | Logs everything in BASIC, plus all the request and response body.    |
| body_and_headers | Logs everything in HEADERS and BODY.                                 |

**NOTE**: When logging the body of request and response, please ensure that they do not contain confidential
information. When logging headers, the client library has a default set of headers that are considered safe to log
but this set can be updated by updating the log options in the builder as shown below:

```java
clientBuilder.httpLogOptions(new HttpLogOptions().addAllowedHeaderName("safe-to-log-header-name"))
```

### Authentication errors

> Azure Monitor Query supports Azure Active Directory authentication. Both LogsQueryClientBuilder and
> MetricsQueryClientBuilder have methods to set the `credential`. To provide a valid credential, you can use

Reviewer: ```suggestion
Azure Monitor Query supports Azure Active Directory authentication. MetricsClientBuilder has a method to set the `credential`. To provide a valid credential, you can use
```

`azure-identity` dependency. For more details on getting started, refer to

> the [README](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/monitor/azure-monitor-query#create-the-client)

Reviewer: Fix links

Reviewer: This needs to be fixed
